test/dynamo/test_misc.py

```diff
@@ -7785,6 +7786,32 @@ def forward(self, x, y):
 
         self.assertEqual(counter.frame_count, 1)
 
+    def test_dynamic_sources_precedence_over_int_specialization(self):
+        builder._DYNAMIC_SOURCES = None
+
+        counter = CompileCounter()
+
```

Comment:
use     @torch.compiler.config.patch(dynamic_sources=...) otherwise those will be set forever in the test

Reply:
Added, still have the _DYNAMIC_SOURCES set to None to clear the cache.